### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by whu-lee** [ajhamdi/MVTN#2]
"killed"   will appear in second epoch of code training.

![1631711289(1)](https://user-images.githubusercontent.com/69310975/133438991-ed441025-8210-4113-bd87-5a5e292d0ca1.jpg)

**Comment by ajhamdi** [comment]
@whu-lee could you please explain more? What is causing this in the code? what are the potential suspects?

**Comment by whu-lee** [comment]
I'm sorry, I do not have a clear description of the problem. 

In the second cycle of training, the program will be interrupted inexplicably, and the terminal displays "killed". I check the memory and it shows that the memory has been completely used, and I suspect it is the cause of the memory leak.

It may be caused by the calculation of loss during training. I changed "total_loss += loss" to "total_loss += loss.item()".There is no memory leak at present.

By the way , I add "torch.cuda.empty_cache()" for the problem of excessive memory usage of cuda.

**Comment by ajhamdi** [comment]
@whu-lee 
I updated the code as per your recommendations. Please let me know if the problem persists.
